pynbody/halo/adaptahop.py

```diff
@@ -102,9 +102,7 @@ def __init__(self, sim, fname=None, read_contamination=None, longint=None):
         # Call parent class
         super(BaseAdaptaHOPCatalogue, self).__init__(sim)
 
-        # Initialize internal data
-        self._base_dm = sim.dm
-
+        self._sim = sim  # prevent garbage collection
```

Comment:
I suspect there is a reason why the original class stored `sim.dm` rather than `sim` itself. Specifically, `pynbody` recreates a new object every time `sim.dm` is called, which may be inefficient and/or lead to confusion (since generally `sim.dm is sim.dm` may evaluate as `False`!)

Possible that this doesn't actually matter here, but just wanted to check it had been thought through.